### PR TITLE
Add autocorrect for ReturnArrowWhitespaceRule

### PR DESCRIPTION
**Pull request by CraigSiemens**

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/ReturnArrowWhitespaceRule.swift [hunk 1]
```diff
@@ -37,20 +37,81 @@ public struct ReturnArrowWhitespaceRule: ConfigurationProviderRule {
...
+        let description = self.dynamicType.description
+        var corrections = [Correction]()
+        var contents = file.contents
+
+        let results = matches.reverse().flatMap { range in
+            return regularExpression.firstMatchInString(contents, options: [], range: range)
+        }
+
+        for result in results {
+            guard result.numberOfRanges == 8 else { break }
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/ReturnArrowWhitespaceRule.swift]: would be good to add a comment why 8 is the magic number here

**Comment by jpsim** [comment]
This looks great @CraigSiemens, could you please add a changelog entry following the guidelines in our CONTRIBUTING.md document?

**Comment by CraigSiemens** [comment]
Good point on the magic number. I changed it to be based on the largest replacement index that it will be expecting.

**Comment by jpsim** [comment]
:+1: sweeeet! Big thanks for this @CraigSiemens!

**Pull request merged**